number_theory/numeric.h

```diff
@@ -43,6 +45,54 @@ std::pair<T, T> exgcd(T a, T b) {
   return std::make_pair(xa, ya);
 }
 
+// Accumulates values according to the binary representation of |binary|.
+// For each bit of |binary|, from lower to higher, it calls |update| to
+// accumulate the values. The initial value is |init|.
+template <class T, class U>
+U binary_accumulate(T binary,
```

Comment:
What about naming `init` as `initial_value` and `update` as `f` or `function` and say that `function` is applied to each bit? Update sounds a little bit confusing.